coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/core/ProposalDevelopmentDocument.java

```diff
@@ -611,8 +611,8 @@ public NoteType getNoteType() {
 
     @Override
     public List getNotes() {
-        if (CollectionUtils.isEmpty(notes)) {
-            notes = new ArrayList<>(super.getNotes());
+        if (StringUtils.isNotBlank(getNoteTarget().getObjectId())) {
```

Comment:
Yeah, that is what I was thinking too and didn't want to put the fix right away even though I figured what is causing this issue.
I tried different scenarios and nothing worked.
Each time we add a note to the collection and save, we need to get the latest once we persist it.